app/org/sagebionetworks/bridge/services/NotificationsService.java

```diff
@@ -250,18 +251,8 @@ public void deleteRegistration(StudyIdentifier studyId, String healthCode, Strin
     }
     
     public void sendSmsMessage(SmsMessageProvider provider) {
-        checkNotNull(provider);
-        
-        PublishRequest request = provider.getSmsRequest();
-
-        if (request.getMessage().getBytes(Charset.forName("US-ASCII")).length > SMS_CHARACTER_LIMIT) {
-            throw new BridgeServiceException("SMS message cannot be longer than 140 UTF-8/ASCII characters.");
-        }
-        
-        PublishResult result = snsClient.publish(request);
-
-        LOG.debug("Sent SMS message, study=" + provider.getStudy().getStudyIdentifier().getIdentifier()
-                + ", message ID=" + result.getMessageId());
+        // This is now a call-through to SmsService.
+        smsService.sendSmsMessage(provider);
```

Comment:
I'm fine with ripping the bandaid off and rewriting call locations to use smsService directly. It's called from ParticipantService which has all the other service references, since it frequently loads an Account, then calls on to the appropriate service.

Reply:
As per IM discussion, I'd prefer unnecessary code churn in an already large change list. I'd prefer to refactor this out of NotificationService at a later date.